### ISSUE TITLE
Differences to reference implementation

### ISSUE THREAD

**Issue by Gara-Dorta** [JKANG94/WarpGAN-Tensorflow#1]
- [ ]  Binary labels (`label_trg` and `label_org`) should not be fed to the generator
https://github.com/JKANG94/WarpGAN-Tensorflow/blob/7ae66e839b7a9c2a2698d9354367655e30559f88/WarpGAN.py#L227-L228
Only use augmented labels as done here
https://github.com/JKANG94/WarpGAN-Tensorflow/blob/7ae66e839b7a9c2a2698d9354367655e30559f88/WarpGAN.py#L241

- [ ] Apply the smoothness loss only to `w_fake`, not to `w_recon`
https://github.com/JKANG94/WarpGAN-Tensorflow/blob/7ae66e839b7a9c2a2698d9354367655e30559f88/WarpGAN.py#L256

- [ ] Random cropping hurts performance, left right flips is enough
https://github.com/JKANG94/WarpGAN-Tensorflow/blob/7ae66e839b7a9c2a2698d9354367655e30559f88/utils.py#L101

- [ ] I'm not sure about this, but I don't think it's a good idea to use numpy functions as that means that the labels are fixed during training, rather than being sampled on each training step. So remove the first function and replace the second with a tensorflow equivalent
https://github.com/JKANG94/WarpGAN-Tensorflow/blob/7ae66e839b7a9c2a2698d9354367655e30559f88/WarpGAN.py#L240
https://github.com/JKANG94/WarpGAN-Tensorflow/blob/7ae66e839b7a9c2a2698d9354367655e30559f88/ops.py#L174
To avoid `h` exploding when all the values are zero you can do
  ```python
  hat_c_norm = tf.norm(hat_c, ord=1, axis=1)
  h = tf.math.divide_no_nan(r, hat_c_norm)
  ```

**Comment by JKANG94** [comment]
I used the edited label as the input of the generator, hat_c  and -hat_c. 
dist = tf_contrib.distributions.Categorical(probs=[0.25, 0.5, 0.25])
hat_c = tf.cast(dist.sample([self.batch_size, self.c_dim]) - 1, dtype ='float32')

x_fake, w_fake = self.generator(self.x_real, hat_c) # real a
x_recon, w_recon = self.generator(x_fake, -hat_c, reuse=True) # real b

**Comment by Gara-Dorta** [comment]
Those changes look good to me. I'd say the only think left is to centre crop the images before the resize to 128x128, as done here.
https://github.com/yunjey/stargan/blob/30867d6f85a3bb99c38ae075de651004747c42d4/data_loader.py#L77
This boost performance a bit, specially for StarGAN, as the model can ignore more of the background. Also it ensures that images keep the original aspect ratio, as they'll be resized from 178x178 to 128x128.

**Comment by JKANG94** [comment]
The result pictures are same as the input,  and there doesn't seem to be any transformation.

**Comment by Gara-Dorta** [comment]
Not surprisingly, if the generator is penalized on the real image logits rather than the fake ones
https://github.com/JKANG94/WarpGAN-Tensorflow/blob/b6724d8ea6869459ee68e465cdbb2e96fc69a214/WarpGAN.py#L248
Should be
```python
g_cls_loss = binary_label_loss(hat_c, fake_cls, self.c_dim) 
```

**Comment by Gara-Dorta** [comment]
A minor point on performance
https://github.com/JKANG94/WarpGAN-Tensorflow/blob/b6724d8ea6869459ee68e465cdbb2e96fc69a214/ops.py#L177
Can be substituted by
```python
bar_c = tf.maximum(hat_c, 0)
```
